src/averageAges.js

```diff
@@ -55,7 +80,24 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const child = people.filter(children => onlyWithSon
+    ? people.some(
+      person => children.mother === person.name
+    )
+    && children.sex === 'm'
+    : people.some(
+      person => children.mother === person.name
```

Comment:
Здесь тоже дублируется код, можно вынести это в отдельную переменную, например.

Reply:
Синтаксически не вижу смысла. Длина кода не изменится. На усмотрение ментора.

Reply:
Again, Olya is right. Not length but reusability matters. Here, you've got a small task, so it may seem appropriate to you, but in other cases, you'll struggle by avoiding this approach.
The lesson is – don't repeat yourself